claims.go

```diff
@@ -8,8 +8,15 @@ import (
 
 // For a type to be a Claims object, it must just have a Valid method that determines
 // if the token is invalid for any supported reason
+// opts will often be nil
 type Claims interface {
-	Valid() error
+	Valid(opts *ValidationOptions) error
+}
+
+// Options passed in to Claims.Valid
+// Currently only supports Leeway (more coming soon)
+type ValidationOptions struct {
+	Leeway int64 // allow a bit (a minute or so) of extra time to allow for clock sku
```

Comment:
Consider naming it "LeewaySeconds" to make it clear that it should be specified in seconds

Reply:
a `time.Duration` might even be better.